akka-http-core/src/main/scala/akka/http/impl/model/parser/CommonRules.scala

```diff
@@ -403,7 +406,13 @@ private[parser] trait CommonRules { this: Parser with StringBuilding ⇒
   private def digitInt(c: Char): Int = c - '0'
   private def digitLong(c: Char): Long = (c - '0').toLong
 
-  private def createDateTime(year: Int, month: Int, day: Int, hour: Int, min: Int, sec: Int, wkday: Int) = {
+  private def createDateTime(rawYear: Int, month: Int, day: Int, hour: Int, min: Int, sec: Int, wkday: Int) = {
```

Comment:
If you add the year expansion logic here the additional comparisons will be run *everytime*, even for the large majority of cases where we matched a 4-digit year and know that we will always hit the default branch.
If you add the year expansion logic to the end of the `date2` rule it will only run when actually required.
Additional benefit: you'll only need a single comparison (`_ <= 69`) instead of 4.